compiler/sighashes.nim

```diff
@@ -179,6 +179,9 @@ proc hashType(c: var MD5Context, t: PType; flags: set[ConsiderFlag]) =
           # make the hash different from the one we produce by hashing only the
           # type name.
           c &= ".empty"
+      else:
+        if sfCompilerProc notin t.sym.flags:
+          c &= t.id
```

Comment:
It we use the id regardless there is no reason to do this convoluted way of doing things. Instead `hashSym` needs to take into account "module1" vs "module2" but that doesn't work either. Not sure what to do here.

Reply:
before this module1 and module2 just hash its type kind, without `if` condition will produce duplicated TNimType, that's all I get.

Reply:
Could the lineInfo's `FileIndex` not be used, or is that not temporally consistent enough?

Reply:
may use hashsym hash type here,at least mitigates the problem

Reply:
Not consistent enough for my taste. There is a better way but it's pretty intrusive...